nova_plugin/server.py

```diff
@@ -253,7 +253,12 @@ def rename(name):
     ctx.logger.debug(
         "server.create() server before transformations: {0}".format(server))
 
-    _handle_image_or_flavor(server, nova_client, 'image')
+    if 'block_device_mapping' in server:
+        # if there is a connected boot volume, don't require the `image`
+        # property.
+        server['image'] = ''
```

Comment:
Why don't we just validate that if both block device mapping and image were provided or let OpenStack do the validation?